File: backend/app/models/carousel.py
from typing import Optional
from pydantic import BaseModel

class CarouselBannerBase(BaseModel):
    title: str
    subtitle: Optional[str] = None
    description: Optional[str] = None
    buttonText: Optional[str] = None
    link: Optional[str] = None
    order: int = 0
    is_active: bool = True

class CarouselBannerCreate(CarouselBannerBase):
    pass

class CarouselBannerUpdate(CarouselBannerBase):
    title: Optional[str] = None

class CarouselBanner(CarouselBannerBase):
    id: int
    image: Optional[str] = None

File: app/src/schemas/carousel.ts
import { z } from "zod";
import { AuditSchema } from "./base";

export const CarouselBannerSchema = z
    .object({
        id: z.number(),
        title: z.string(),
        subtitle: z.string().optional(),
        description: z.string().optional(),
        buttonText: z.string().optional(),
        image: z.string(),
        link: z.string().optional(),
        order: z.number(),
        is_active: z.boolean(),
    })
    .merge(AuditSchema);

export type CarouselBanner = z.infer<typeof CarouselBannerSchema>;
